src/experimental/patterns/ec2-app.ts

```diff
@@ -40,14 +59,11 @@ class AutoScalingRollingUpdateTimeout implements IAspect {
       const currentRollingUpdate = construct.cfnOptions.updatePolicy?.autoScalingRollingUpdate;
       const currentCreationPolicy = construct.cfnOptions.creationPolicy;
 
-      /**
-       * The type of `healthCheckGracePeriod` is `number | undefined`.
-       * In reality, it will always be set as we set it in {@link GuEc2App}.
-       * The right-hand side to appease the compiler; 5 minutes is the default value used by AWS.
-       *
-       * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-updatepolicy.html#cfn-attributes-updatepolicy-rollingupdate-pausetime
-       */
-      const signalTimeoutSeconds = construct.healthCheckGracePeriod ?? Duration.minutes(5).toSeconds();
+      if (!construct.healthCheckGracePeriod) {
+        throw new Error(`The healthcheck grace period not set for autoscaling group ${construct.node.id}.`);
+      }
```

Comment:
The `healthCheckGracePeriod` is typed `number | undefined`. When `undefined`, [the CloudFormation default of 0 seconds is used](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-autoscaling-autoscalinggroup.html#cfn-autoscaling-autoscalinggroup-healthcheckgraceperiod).

Our user data will at least download the application artifact from S3. This'll take longer than 0 seconds. Furthermore, the [`GuEc2App` pattern defaults it to 2 minutes](https://github.com/guardian/cdk/blob/3d81b9215243419e496ef4a71f20ed9e3986a1a5/src/patterns/ec2-app/base.ts#L428). That is, we can realistically expect this property to always be defined.